### PR TITLE
Add click track to BasePlayer class

### PR DESCRIPTION
**Pull request by psc-g**
Adds two "click track" sound patterns to the default Drum pitch classes, and adds parameter to the `start` method to include a click track.
Added demo functionality to music_vae.html.

### CODE REVIEW

File: music/src/core/player.ts [hunk 1]
```diff
@@ -52,7 +62,31 @@ export abstract class BasePlayer {
    * default of 120. Only valid for quantized sequences.
    * @returns a Promise that resolves when playback is complete.
    */
-  start(seq: INoteSequence, qpm?: number): Promise<void> {
+  start(seq: INoteSequence, click?: boolean, qpm?: number): Promise<void> {
```
Comment (adarob) [inline music/src/core/player.ts]: this is not backward compatible

File: music/src/core/data.ts [hunk 2]
```diff
@@ -200,6 +200,11 @@ export class DrumsConverter extends DataConverter {
         this.pitchToClass.set(p, c);
       });
     }
+    // We also add two extra pitches for the click-track.
+    // lo-click.
+    this.pitchToClass.set(89, this.pitchClasses.length + 1);
```
Comment (adarob) [inline music/src/core/data.ts]: Let's move this to constants.ts.

**Comment by psc-g** [comment]
thanks adam, talp?

File: music/src/core/player.ts [hunk 3]
```diff
@@ -175,6 +199,8 @@ class DrumKit {
         this.DRUM_PITCH_TO_CLASS.set(p, c);
       });
     }
+    this.DRUM_PITCH_TO_CLASS.set(89, 9);
```
Comment (adarob) [inline music/src/core/player.ts]: Use constants.

File: music/src/core/data.ts [hunk 4]
```diff
@@ -200,6 +200,11 @@ export class DrumsConverter extends DataConverter {
         this.pitchToClass.set(p, c);
       });
     }
+    // We also add two extra pitches for the click-track.
```
Comment (adarob) [inline music/src/core/data.ts]: Why are these needed in the DataConverter?

Comment (psc-g) [reply music/src/core/data.ts]: old attempts, sorry, removed.

File: music/src/core/player.ts (hunk 3 above)
Comment (psc-g) [reply music/src/core/player.ts]: done
